### ISSUE TITLE
[Accessibility]: Code Insights Edit UI - Unchecking all repos does not announce that it unchecked

### ISSUE THREAD

**Issue by unclejustin** [sourcegraph/sourcegraph#41686]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33501

### Problem description

![PixelSnap 2022-09-15 at 13 08 57](https://user-images.githubusercontent.com/1855233/190478362-7773d02a-5902-4f48-bcd9-40dca8b05b7c.png)

When unchecking the "Run your insight..." checkbox, it clears the repositories input. This makes the input invalid. The voice-over immediately announces that the input is invalid and required, but not why that happened.

### Expected behavior

The voice-over should announce that the checkbox is unchecked first.

Then should announce that the repositories input was cleared.

Then continue as before.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by vovakulikov** [comment]
I think we have an issue about fixing this behaviour around unchecking and clearing repositories input. We should preserve repositories input when we check/uncheck all repositories checkbox. I think this will solve this problem with screen reader announcements
